src/utils/cache.ts

```diff
@@ -27,6 +28,25 @@ class CachedAnswers {
     this.lru = hashlru(maxSize)
   }
 
+  /**
+   * We index the cache on 'domain-{@link RecordType}` instead of
+   * 'domain-{@link RecordType}|{@link RecordTypeLabel}' to ensure that we don't
+   * cache the same answers separately for RecordType.A and RecordTypeLabel.A.
+   *
+   * This means, if you query a resolver with `useRecordTypeValue=true`, and
+   * they return an empty answer, that empty answer will be cached, and a
+   * subsequent call with `useRecordTypeValue=false` would need to be paired
+   * with `cached: false` to avoid getting the empty answer back when using a
+   * {@link RecordTypeLabel}.
+   *
+   * NOTE: this will resolve/obfuscate the issue where dns resolvers return
+   * different answers depending on the value of the "type" field in the query.
+   * But give the user the ability to retry with a different type if they want
+   */
+  private getKey (domain: string, type: RecordType | RecordTypeLabel): string {
+    return `${domain.toLowerCase()}-${convertType(type, true)}`
+  }
```

Comment:
Are we good with this? We could also check the cache for either cached type in `dns.ts` and then convert it, so we're only caching exactly what the response is.

i.e. 

```
    let cached
    if (options.cached !== false) {
      cached = this.cache.get(domain, types) ?? this.cache.get(domain, getTypes(types, !options.useRecordTypeValue))
    }
```